Jl event provider

Jaron-Lane opened the pull request:
# Description
Created the EventDataProvider.js. It references the database.json/events and exports the getEvents and useEvents functions. This branch also imports and calls getEvents in Nutshell.js. Should console.table the data from the provider. 

Contains broken code from my upcoming branch inside EventButton.js. Irrelevant to current feature.

## Type of change
- [x] New feature (non-breaking change which adds functionality)

# Testing Instructions
`git fetch --all`
`git checkout jl-event-provider`

Should only display table of the data from the EventDataProvider to the console.

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings or errors
- [x] I have added test instructions that prove my fix is effective or that my feature works

Jaron-Lane committed:
first version of EventDataProvider, without url and without testing

Jaron-Lane committed:
updated database fetch

Jaron-Lane committed:
Merge branch 'master' into jl-event-provider

Jaron-Lane committed:
Imported and called EventDataProvider inside Nutshell.js

Jaron-Lane committed:
added comments throughout feature

jisie commented:
@Jaron-Lane What should we see when we test this feature? Should data be rendered on the DOM? Are we looking for a `console.log()`? Add a description of what we need to look for when we test your code to the pull request.

Jaron-Lane committed:
uncommented the console.table

Jaron-Lane commented:
> @Jaron-Lane What should we see when we test this feature? Should data be rendered on the DOM? Are we looking for a `console.log()`? Add a description of what we need to look for when we test your code to the pull request.

Just changed the code to display a table of the event data to the console. Also changed the description to match your specifications. Please let me know if this is what you meant :)

tedmarov reviewed (commented):
I see your console table in dev tools 👍🏼

TrentSisson reviewed (dismissed):
I see your console.table in the console
looks good

Jaron-Lane committed:
fixing merge conflict with index.html and nutshell.js
